src/Interpreters/JIT/CHJIT.cpp

```diff
@@ -490,19 +505,52 @@ void CHJIT::runOptimizationPassesOnModule(llvm::Module & module) const
     llvm::CGSCCAnalysisManager cgam;
     llvm::ModuleAnalysisManager mam;
 
+    auto target_analysis = machine->getTargetIRAnalysis();
```

Comment:
Future readers will not understand what 508-509 is doing and why. What kind of "pass" do we register here? This needs a comment.

Reply:
EDIT: I found similar code [here](https://www.neilhenning.dev/posts/llvm-new-pass-manager/) - it would still be nice to add a comment in case you have some more info. If not, then we can leave it as is.

Reply:
OK, I will add some comment.